### PR TITLE
sink: add a method to validate sink

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

When walking the sink code, I found some duplicate code for validating the health of the sink, this PR try to make it dry.

### What is changed and how it works?

add a new method to the sink package.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by amyangfei — APPROVED** [review]

**Review by ben1009 — APPROVED** [review]

File: cdc/sink/sink.go [hunk 1]
```diff
@@ -118,3 +118,29 @@ func NewSink(ctx context.Context, changefeedID model.ChangeFeedID, sinkURIStr st
 	}
 	return nil, cerror.ErrSinkURIInvalid.GenWithStack("the sink scheme (%s) is not supported", sinkURI.Scheme)
 }
+
+// Validate the health of the sink
```
Comment (asddongmen) [inline cdc/sink/sink.go:122]: In fact, what we are checking is whether sinkuri and config are valid.
So this comment may be better?  
```
Validate whether the sinkuri and config are valid.
```

**Comment by 3AceShowHand** [comment]
@amyangfei I think we should also cherry-pick this to other branches, what do you think?

**Pull request merged**